xbmc/cores/AudioEngine/Utils/PackerMAT.cpp

```diff
@@ -57,42 +57,27 @@ CPackerMAT::CPackerMAT()
 // high-bitrate streams can overshoot this size and therefor require proper handling of dynamic padding.
 bool CPackerMAT::PackTrueHD(const uint8_t* data, int size)
 {
-  TrueHDMajorSyncInfo info;
-
-  // get the ratebits and output timing from the sync frame
+  // get the ratebits from the sync frame
   if (AV_RB32(data + 4) == FORMAT_MAJOR_SYNC)
   {
-    info = ParseTrueHDMajorSyncHeaders(data, size);
-
-    if (!info.valid)
+    if (size < 32)
```

Comment:
The thing with the simpification is, that we now have "no speaking"  values, e.g.  reader needs to know what 32 means in this context and below what data[8] >> 4 would mean, etc.

Reply:
Improved based on this:

![thd-spec](https://github.com/xbmc/xbmc/assets/58434170/57775690-1fd6-45f1-8e98-17fe618cf08a)

`format_sync` is the magic 32 bits = 0xf8726fba

and immediately follows `format_info` (others 32 bits) and the `audio_sampling_frequency` field is the high nibble of first byte.

The previous code (based on FFmpeg) is correct but redundant...

```c++
    if (data[7] == 0xba)
      m_state.ratebits = data[8] >> 4;
    else if (data[7] == 0xbb)
      m_state.ratebits = data[9] >> 4;
    else
      return false;
```

`0xba` is the end of format_sync and is redundant because is already compared.

`0xbb` code path is not possible for the same reason.

`return false` is also not possible code path.